consensus/spos/bls/subroundEndRound.go

```diff
@@ -195,17 +195,14 @@ func (sr *subroundEndRound) doEndRoundJobByLeader() bool {
 		return false
 	}
 
-	multiSigner, err := sr.MultiSignerContainer().GetMultiSigner(sr.Header.GetEpoch())
+	err = sr.SignatureHandler().SetMultiSignerByEpoch(sr.Header.GetEpoch())
 	if err != nil {
 		log.Error("doEndRoundJobByLeader.GetMultiSigner", "error", err.Error())
 		return false
 	}
 
-	// TODO: get the signatures from consensus state
-	pubKeysSigners, sigShares := sr.getSigningData()
-
 	// Aggregate sig and add it to the block
-	sig, err := multiSigner.AggregateSigs(pubKeysSigners, sigShares)
+	sig, err := sr.SignatureHandler().AggregateSigs(bitmap)
```

Comment:
you can add the epoch as parameter to AggregateSigs and remove the Set from above.

Reply:
added epoch parameter